tutorials/10_DebuggingREMIND.md

```diff
@@ -0,0 +1,169 @@
+
+More information needed? debug runs
+--------------------------
+
+If you would like GAMS to print specific information, add a `display` statement as close to the solve statement as possible, i.e., in presolve, or right after the data is loaded in datainput, and use `Rscript start.R --reprepare` to regenerate the `full.gms` file and restart the model, or just change `full.gms` directly and rerun the folder directly.
+
+If this is not sufficient, you may start a "debug" run which is more generous in logging, but runs on one single node only instead of the parallel mode with one node per region. Therefore, they take a lot of time and the log files are substantially increased to GB of data.
+
+To restart a run in "debug" mode, run
+```bash
+Rscript start.R --debug --reprepare
+```
+in the REMIND main folder and select the runs to be restarted (the shortcut is `Rscript start.R -dR`). This will recreate move `full.gms` and `fulldata.gdx` to filenames with an appended `_old` and restart the input preparation and the model run.
+
+If you only want to look at a specific region that created the infeasibility, you can either specify `c_testOneRegi_region` in `scenario_config_XYZ.csv`:
+```bash
+Rscript start.R --debug --interactive --testOneRegi scenario_config_XYZ.csv
+```
+and you will be asked which region you want to look at (the shortcut is `Rscript start.R -di1` + the file name).
+
+If the error was not in the first iteration, a faster and more convenient way might be to start the run again in parallel mode with `c_keep_iteration_gdxes = 1` specified in `config/default.cfg` or the `scenario_config_XYZ.csv`. And then start a debug run that uses the `fulldata_07.gdx` (or the first iteration in which this occurs) of this first run as `input.gdx` specified in `path_gdx` of `scenario_config_XYZ.csv`. That way, you should see where the infeasibility is in the first iteration of the debug run already and not have to wait until the 7th iteration.
+
+If the iterations are not sufficient to converge, you can run REMIND for more iterations by modifying `cm_iteration_max` in [`80_optimization/nash/datainput.gms`](../modules/80_optimization/nash/datainput.gms) and the set iteration in [`core/sets.gms`](../core/sets.gms), which by default only goes to 200.
+
+If you want a closer look on the GAMS CONOPT output during a REMIND run, you can access the solver logs by moving to the output folder an running:
+
+```bash
+find -name gmsgrid.log | xargs tail -n 12
```

Comment:
Use `conoptspy` instead.

Reply:
No, they don't have to.  It's not like they are buying a product with a customer support agreement.  In the meantime we are burdening ourselves with thinking about hypothetical unicorns that could use REMIND in their time-travelling flying saucers …